cmake-modules/FolderList.cmake

```diff
@@ -2,38 +2,39 @@ macro(GetFolderList project)
   
     if(${project} STREQUAL CERTIFICATES)
-        DownloadDepVersion(sdk/core azure-core 1.3.1)
+        DownloadDepVersion(sdk/core azure-core 1.2.0)
         DownloadDepVersion(sdk/identity azure-identity 1.1.0)
+    elseif(${project} STREQUAL IDENTITY)
+        DownloadDepVersion(sdk/core azure-core 1.2.0)
     endif()
 
     list(REMOVE_DUPLICATES BUILD_FOLDERS)
 endmacro()
 
 macro(SetCompileOptions project)
     message ("setting up compile options for ${project}")
-    
-    if(${project} STREQUAL CERTIFICATES)
+   
     # Compile Options
-        option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
-        option(BUILD_TRANSPORT_CURL "Build an HTTP transport implementation with CURL" OFF)
-        option(BUILD_TRANSPORT_WINHTTP "Build an HTTP transport implementation with WIN HTTP" OFF)
-        option(BUILD_TRANSPORT_CUSTOM "Implementation for AzureSdkGetCustomHttpTransport function must be linked to the final application" OFF)
-        option(BUILD_TESTING "Build test cases" OFF)
-        option(BUILD_RTTI "Build libraries with run-time type information." ON)
-        option(BUILD_CODE_COVERAGE "Build gcov targets for HTML and XML reports. Requires debug build and BUILD_TESTING" OFF)
-        option(BUILD_DOCUMENTATION "Create HTML based API documentation (requires Doxygen)" OFF)
-        option(RUN_LONG_UNIT_TESTS "Tests that takes more than 5 minutes to complete. No effect if BUILD_TESTING is OFF" OFF)
-        option(BUILD_PERFORMANCE_TESTS "Build the performance test library" OFF)
-        option(MSVC_USE_STATIC_CRT "(MSVC only) Set to ON to link SDK with static CRT (/MT or /MTd switch)." OFF)
-        option(BUILD_FROM_SOURCE_EXPORT "if export is built from source" ON)
-        option(TESTING_BUILD "BUILD test cases" OFF)
-    endif()
+    option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
+    option(BUILD_TRANSPORT_CURL "Build an HTTP transport implementation with CURL" OFF)
+    option(BUILD_TRANSPORT_WINHTTP "Build an HTTP transport implementation with WIN HTTP" OFF)
+    option(BUILD_TRANSPORT_CUSTOM "Implementation for AzureSdkGetCustomHttpTransport function must be linked to the final application" OFF)
+    option(BUILD_TESTING "Build test cases" OFF)
+    option(BUILD_RTTI "Build libraries with run-time type information." ON)
+    option(BUILD_CODE_COVERAGE "Build gcov targets for HTML and XML reports. Requires debug build and BUILD_TESTING" OFF)
+    option(BUILD_DOCUMENTATION "Create HTML based API documentation (requires Doxygen)" OFF)
+    option(RUN_LONG_UNIT_TESTS "Tests that takes more than 5 minutes to complete. No effect if BUILD_TESTING is OFF" OFF)
+    option(BUILD_PERFORMANCE_TESTS "Build the performance test library" OFF)
+    option(MSVC_USE_STATIC_CRT "(MSVC only) Set to ON to link SDK with static CRT (/MT or /MTd switch)." OFF)
+    option(BUILD_FROM_SOURCE_EXPORT "if export is built from source" ON)
+    option(TESTING_BUILD "BUILD test cases" OFF)
```

Comment:
make this a cmake macro and call it from the root cmakelist and from here.  That way we have all options in one single place.

Reply:
there might be some diffrences over time. i'd rather have it here